mmdet/models/data_preprocessors/data_preprocessor.py

```diff
@@ -552,3 +553,177 @@ def cpu(self, *args, **kwargs) -> nn.Module:
         """
 
         return self.data_preprocessor.cpu(*args, **kwargs)
+
+
+@MODELS.register_module()
+class BatchResize(nn.Module):
+    """Batch resize during training.
+
+    Args:
+        scale (tuple): Images scales for resizing.
+        pad_size_divisor (int): Image size divisible factor.
+            Defaults to 1.
+    """
+
+    def __init__(self, scale, pad_size_divisor=1) -> None:
+        super().__init__()
+        self.min_size = min(scale)
+        self.max_size = max(scale)
+        self.pad_size_divisor = pad_size_divisor
+
+    def forward(
+        self, inputs: Tensor, data_samples: List[DetDataSample]
+    ) -> Tuple[Tensor, List[DetDataSample]]:
+        """resize a batch of images and bboxes."""
+
+        batch_height, batch_width = inputs.shape[-2:]
+        t_height, t_width, scale = self.target_size(batch_height, batch_width)
+
+        device = inputs.device
+        padded_images = inputs.cpu().detach().numpy().transpose(0, 2, 3, 1)
+        resized_images = np.array(
+            [mmcv.imrescale(im, (t_width, t_height)) for im in padded_images])
```

Comment:
why not use torch interpolate?

Reply:
The  implementation has been replaced with torch interpolater.